### ISSUE TITLE
[Bug]: Mapped TOR Domain leads to blank page at checkout

### ISSUE THREAD

**Issue by petzsch** [btcpayserver/btcpayserver#6019]
### What is your BTCPay version?

BTCPay Server v1.13.1+ca55e1f30

### How did you deploy BTCPay Server?

docker

### What happened?

I've created a domain mapping for my test PoS app for the tor domain:
![grafik](https://github.com/btcpayserver/btcpayserver/assets/1374810/79b504d2-3830-41e9-8b1c-4ba6b26e8297)

I accessed the PoS App via the .onion URL:
![grafik](https://github.com/btcpayserver/btcpayserver/assets/1374810/f523637b-906c-49fa-b76f-d2bf4c7d9511)

I click on the Buy button for the first item and end on a white page:
![grafik](https://github.com/btcpayserver/btcpayserver/assets/1374810/f9ebbc95-efe4-4132-8d17-ede5b6c7e88a)

In the server log I find the attached error message:

### How did you encounter this bug?

see above

### Relevant log output

```shell
2024-05-30 15:42:01.198 +00:00 [ERR] Connection id "0HN40FBR16FDT", Request id "0HN40FBR16FDT:00000001": An unhandled exception was thrown by the application.
System.ArgumentException: An element with the key 'appId' already exists in the RouteValueDictionary. (Parameter 'key')
   at Microsoft.AspNetCore.Routing.RouteValueDictionary.Add(String key, Object value)
   at BTCPayServer.Filters.DomainMappingConstraintAttribute.Accept(ActionConstraintContext context) in /source/BTCPayServer/Filters/DomainMappingConstraintAttribute.cs:line 64
   at Microsoft.AspNetCore.Mvc.Routing.ActionConstraintMatcherPolicy.EvaluateActionConstraintsCore(HttpContext httpContext, CandidateSet candidateSet, IReadOnlyList`1 items, Nullable`1 startingOrder)
   at Microsoft.AspNetCore.Mvc.Routing.ActionConstraintMatcherPolicy.EvaluateActionConstraints(HttpContext httpContext, CandidateSet candidateSet)
   at Microsoft.AspNetCore.Mvc.Routing.ActionConstraintMatcherPolicy.ApplyAsync(HttpContext httpContext, CandidateSet candidateSet)
   at Microsoft.AspNetCore.Routing.Matching.DfaMatcher.SelectEndpointWithPoliciesAsync(HttpContext httpContext, IEndpointSelectorPolicy[] policies, CandidateSet candidateSet)
   at Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware.<Invoke>g__AwaitMatch|10_1(EndpointRoutingMiddleware middleware, HttpContext httpContext, Task matchTask)
   at BTCPayServer.Hosting.BTCPayMiddleware.Invoke(HttpContext httpContext) in /source/BTCPayServer/Hosting/BTCpayMiddleware.cs:line 97
   at BTCPayServer.Hosting.GreenfieldMiddleware.Invoke(HttpContext httpContext) in /source/BTCPayServer/Hosting/GreenfieldMiddleware.cs:line 30
   at Microsoft.AspNetCore.Diagnostics.StatusCodePagesMiddleware.Invoke(HttpContext context)
   at BTCPayServer.Hosting.HeadersOverrideMiddleware.Invoke(HttpContext httpContext) in /source/BTCPayServer/Hosting/HeadersOverrideMiddleware.cs:line 29
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)
```

### What browser do you use?

TorBrowser

### Additional information

none

### Are you sure this is a bug report?

- [X] I confirm this is a bug report

**Comment by pavlenex** [comment]
@Kukks  @dennisreimann do you guys know if we support Tor domains here, it seems there's log output so we can use it to fix this?

**Comment by dennisreimann** [comment]
I'm currently debugging it and think the exact version doesn't matter here. Will provide a fix shortly, which @petzsch can then deploy and test, because we cannot really verify the fix in the dev env.

**Comment by petzsch** [comment]
The issue is not reproducable in 1.13.3
marking this issue as solved.

**Comment by dennisreimann** [comment]
Yup, my bad! Saw I fixed it already in #5948, when I wanted to fix it again 😆
